ee/tabby-webserver/src/service/auth.rs

```diff
@@ -301,11 +286,40 @@ impl AuthenticationService for DbConn {
         }
     }
 
-    async fn create_invitation(&self, email: String) -> Result<Invitation> {
-        let invitation = self.create_invitation(email).await?;
+    async fn create_invitation(
+        &self,
+        email: String,
+        mail_service: Arc<dyn EmailService>,
```

Comment:
instead of passing this through argument, I suggest create an wrapper struct of AuthenticationService, and create it with email service embed.
```rust
struct AuthenticationServiceImpl {
  db: DbConn,
  mail: Arc<dyn EmailService>
}
```